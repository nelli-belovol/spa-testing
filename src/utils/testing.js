import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom"; // MemoryRouter was special created for testing 


export const renderWithRouter=(component, options={})=>{
	const Wrapper = ({children})=>(<MemoryRouter {...options}>{children}</MemoryRouter>)

	render(component, {wrapper: Wrapper})
}