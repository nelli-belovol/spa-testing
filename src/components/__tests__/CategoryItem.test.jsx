import { screen } from "@testing-library/react";

import { renderWithRouter } from "../../utils/testing";

import { CategoryItem } from "../CategoryItem";

describe("CategoryItem", () => {
  it("should render correctly", () => {
    renderWithRouter(
      <CategoryItem
        strCategory="Cake"
        strCategoryThumb="Cake.png"
        strCategoryDescription="Chocolate cake is a cake flavored with melted chocolate, cocoa powder, or both."
      />,
    );
    expect(screen.getByRole("article")).toMatchSnapshot();
  });
});
