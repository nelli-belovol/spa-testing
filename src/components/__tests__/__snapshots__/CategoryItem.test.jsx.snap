// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem should render correctly 1`] = `
<div
  class="card"
  role="article"
>
  <div
    class="card-image"
  >
    <img
      alt="Cake"
      src="Cake.png"
    />
  </div>
  <div
    class="card-content"
  >
    <span
      class="card-title"
    >
      Cake
    </span>
    <p>
      Chocolate cake is a cake flavored with melted chocolate, coc
      ...
    </p>
  </div>
  <div
    class="card-action"
  >
    <a
      class="btn"
      href="/category/Cake"
    >
      Watch category
    </a>
  </div>
</div>
`;
